nifi-nar-bundles/nifi-atlas-bundle/nifi-atlas-reporting-task/src/main/java/org/apache/nifi/atlas/provenance/analyzer/FilePath.java

```diff
@@ -56,11 +59,18 @@ public DataSetRefs analyze(AnalysisContext context, ProvenanceEventRecord event)
             final String hostname = StringUtils.isEmpty(uriHost) ? InetAddress.getLocalHost().getHostName() : uriHost;
             namespace = context.getNamespaceResolver().fromHostNames(hostname);
         } catch (UnknownHostException e) {
-            logger.warn("Failed to get localhost name due to " + e, e);
+            LOGGER.warn("Failed to get localhost name due to " + e, e);
             return null;
         }
 
-        final String path = uri.getPath();
+        final String path;
+        if (context.getFilesystemPathsLevel() == FilesystemPathsLevel.Directory) {
+            final String dirPath = StringUtils.substringBeforeLast(uri.getPath(), PATH_SEPARATOR);
+            path = dirPath.isEmpty() ? PATH_SEPARATOR : dirPath;
+        } else {
+            path = uri.getPath();
+        }
+
         ref.set(ATTR_NAME, path);
         ref.set(ATTR_PATH, path);
```

Comment:
What do you think of extracting these lines to a separate method? The scode fragment is duplicated in HDFSParth:52-61.

Reply:
done